listenbrainz_spark/stats/user/utils.py

```diff
@@ -1,51 +1,48 @@
+from datetime import datetime
 import time
 
 from collections import defaultdict
 from listenbrainz_spark.stats import run_query
+from listenbrainz_spark.utils import get_listens
+from listenbrainz_spark.stats import adjust_months, adjust_days
+from listenbrainz_spark.path import LISTENBRAINZ_DATA_DIRECTORY
+
+
+def get_latest_listen_ts():
+    """ Get the timestamp of the latest timestamp present in spark cluster """
+    now = datetime.now()
+    while True:
+        try:
+            df = get_listens(now, now, LISTENBRAINZ_DATA_DIRECTORY)
+            break
+        except SystemExit:
```

Comment:
we should make `get_listens` raise something else, `SystemExit` here makes no sense.